components/motor/gpio/motor_encoder.go

```diff
@@ -570,104 +422,108 @@ func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, revolutions float
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
+	m.state.regulated = true
 	if revolutions == 0 {
 		// Moving 0 revolutions is a special value meaning "move forever."
 		oldRpm := m.state.desiredRPM
 		m.state.desiredRPM = rpm
-		m.state.regulated = false // we're not going to a position
+		m.state.goalPos = math.Inf(int(rpm))
 
-		if math.Abs(oldRpm) > 0.001 && d == m.directionMovingInLock() {
+		// if we are already moving, let rpmMonitor deal with setPower
+		if math.Abs(oldRpm) > 0.001 && m.state.direction == m.directionMovingInLock() {
 			return nil
 		}
-		err := m.setPower(ctx, float64(d)*.06, true) // power of 6% is random
-		return err
+		// if moving from stop, start at 25% power
+		if err := m.setPower(ctx, m.state.direction*0.25, true); err != nil {
+			return err
+		}
+		return nil
 	}
 
-	numTicks := int64(revolutions * float64(m.ticksPerRotation))
-
-	pos, _, err := m.encoder.Position(ctx, encoder.PositionTypeUnspecified, nil)
+	currentPos, _, err := m.encoder.Position(ctx, encoder.PositionTypeUnspecified, nil)
```

Comment:
Let's make the position type Ticks, this entire loop doesn't work with absolute encoders, and we have one that returns both.